mlflow/server/auth/__init__.py

```diff
@@ -414,20 +418,52 @@ def _get_proxy_artifact_validator(
     }.get(method)
 
 
-@catch_mlflow_exception
-def _before_request():
-    if is_unprotected_route(request.path):
-        return
+def authenticate_request() -> Union[Authorization, Response]:
+    """Use configured authorization function to get request authorization."""
+    auth_func = get_auth_func(auth_config.authorization_function)
+    return auth_func()
+
+
+@functools.lru_cache(maxsize=None)
+def get_auth_func(authorization_function: str) -> Callable[[], Union[Authorization, Response]]:
+    """Import and return the specified authorization function.
+
+    :param authorization_function: A string of the form "module.submodule:auth_func
+    """
+    mod_name, fn_name = authorization_function.split(":")
```

Comment:
```suggestion
    mod_name, fn_name = authorization_function.split(":", 1)
```

Just my habit when unpacking a string split to make the code a little safer.